pkg/joinserver/redis/registry.go

```diff
@@ -121,8 +149,29 @@ func (r *DeviceRegistry) SetByEUI(ctx context.Context, joinEUI types.EUI64, devE
 				return err
 			}
 			f = func(p redis.Pipeliner) error {
-				_, err := ttnredis.SetProto(p, k, stored, 0)
-				return err
+				if _, err := ttnredis.SetProto(p, k, stored, 0); err != nil {
+					return err
+				}
+				uniqueID, err := r.getProvisionerUniqueID(stored)
+				if err != nil {
+					return err
+				}
+				if uniqueID != "" {
```

Comment:
I would prefer `if uniqueID == "" { return nil; }` here to avoid nesting

Reply:
I see what you mean and in general I agree but in this case we're listing a number of actions here. Early exit for a condition that's unrelated to a future action may result in nasty bugs.